docs/rules/custom-rules.md

```diff
@@ -16,49 +22,63 @@ rules:
 mutuallyRequired | [`string`] | Asserts that listed properties (key names only) are mutually required. See [mutuallyRequired example](#mutuallyrequired-example).
+ref | `boolean \| string` | Asserts a reference object presence in object's property. A boolean value of `true` means the property has a `$ref` defined. A boolean value of `false` means the property has not defined a `$ref` (it has an in-place value). A string value means that the `$ref` is defined and the unresolved value must match the pattern (for example, `'/paths\/. *\.yaml$/'`). See [ref example](#ref-example).|
 required | [`string`] | Asserts all listed values are defined. See [required example](#required-example).
 requireAny | [`string`] | Asserts that at least one of the listed properties (key names only) is defined. See [requireAny example](#requireany-example).
-disallowed | [`string`] | Asserts all listed values are not defined. See [disallowed example](#disallowed-example).
-defined | `boolean` | Asserts a property is defined. See [defined example](#defined-example).
-undefined | `boolean` | Asserts a property is undefined. See [undefined example](#undefined-example).
-nonEmpty | `boolean` | Asserts a property is not empty. See [nonEmpty example](#nonempty-example).
-minLength | `integer` | Asserts a minimum length (inclusive) of a string or list (array). See [minLength example](#minlength-example).
-maxLength | `integer` | Asserts a maximum length (exclusive) of a string or list (array). See [maxLength example](#maxlength-example).
-ref | `boolean \| string` | Asserts a reference object presence in object's property. A boolean value of `true` means the property has a `$ref` defined. A boolean value of `false` means the property has not defined a `$ref` (it has an in-place value). A string value means that the `$ref` is defined and the unresolved value must match the pattern (for example, `'/paths\/.*\.yaml$/'`). See [ref example](#ref-example).|
-`{pluginId}/{functionName}` | `object` | Custom assert defined in the plugin. This function will be called with options including the value. See [custom function example](#custom-function-example).|
+`{pluginId}/{functionName}` | `object` | Custom assert defined in the plugin. This function is called with options including the value. See [custom function example](#custom-function-example).
+
+## Where object
 
-## Context object
+The `where` object is part of a `where` list which must be defined in order from the root node.
 
 Property | Type | Description
 -- | -- | --
-type | `string` | **REQUIRED.** One of the [OpenAPI node types](#openapi-node-types).
-matchParentKeys | [`string`] | The list of parent object key names to evaluate with respect to the subject.
-excludeParentKeys | [`string`] | The list of parent object key names to not evaluate with respect to the subject.
-
-See the [context example](#context-example).
-
+whereSubject | Subject object | **REQUIRED.** Filters the subject further based on the definition.
```

Comment:
so you want to add prefix `where`? What is the reason for this? To minimize the confusion?

Reply:
I don't like it personally. Yes, it is to minimize confusion, but I would prefer without it.

Reply:
I removed it in my most recent push.